File: Day04/src/test/java/uk/co/stevebosman/aoc24/day04/MainTest.java
package uk.co.stevebosman.aoc24.day04;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MainTest {

  @Test
  void textMain() {
    assertDoesNotThrow(() -> Main.main(new String[]{"src/test/resources/example1.txt"}));
  }
}

File: Day06/src/test/scala/MainTest.scala
import org.scalatest.funsuite.AnyFunSuite

class MainTest extends AnyFunSuite:
  test("Given straight ahead, part 1") {
    assert(main("src/test/resources/straight.txt") === (2, 0))
  }
  test("Given example, part 1 - strict") {
    assert(main("src/test/resources/example1.txt") === (41, 6))
  }
